src/index.ts

```diff
@@ -85,12 +85,17 @@ const zodToTsNode = (
       const properties = Object.entries(zod._def.shape())
 
       const members: ts.TypeElement[] = properties.map(([key, value]) => {
-        const type = zodToTsNode(value as ZodTypeAny, ...otherArgs)
+        const zodAny = value as ZodTypeAny
+        otherArgs[1].parentIsObject = true
+        const type = zodToTsNode(zodAny, ...otherArgs)
 
+        const flagsRequireOptional = options.treatOptionalsAs === 'both' || options.treatOptionalsAs === 'optional'
```

Comment:
We should try to move this logic outside of the `zodToTsNode` function because it'll rerun this condition unnecessary. This might involve extended the types for `zodToTs` and applying that to `zodToTsNode`.

Reply:
Well, you really are interested in speed :)

I'd leave `ZodToTsOptions` as is, since it's public api, and change `RequiredZodToTsOptions` to `ZodToTsNodeOptions`. So I would calculate whatever internal flags I want (namely `treatOptionalsAsUndefined` and `treatOptionalsAsOptional`) before calling `zodToTsNode`

But I can't make such changes, since I'm not quite sure how `callGetType` works and don't understand why `resolveOptions` is exported to user

I'm usually ok when boolean flag is recalculated and it's not obvious that solution above would be faster. I'd leave the thing as is, otherwise it starts to go further than I expected.

Reply:
Yeah, on second thought, this is fine.